src/plugins/navigation/public/top_nav_menu/top_nav_menu_item.tsx

```diff
@@ -36,27 +40,53 @@ export function TopNavMenuItem(props: TopNavMenuData) {
 
   function handleClick(e: MouseEvent<HTMLButtonElement>) {
     if (isDisabled()) return;
-    props.run(e.currentTarget);
+    if (props.popOverContents) {
+      setPopoverOpen(true);
+    } else if (props.run) {
+      props.run(e.currentTarget);
+    }
   }
 
-  const btn = (
-    <EuiButtonEmpty
-      size="xs"
-      isDisabled={isDisabled()}
-      onClick={handleClick}
-      data-test-subj={props.testId}
-      className={props.className}
-    >
+  const commonButtonProps = {
+    isDisabled: isDisabled(),
+    onClick: handleClick,
+    iconType: props.iconType,
+    'data-test-subj': props.testId,
+  };
+
+  let btn = props.emphasize ? (
+    <EuiButton size="s" fill {...commonButtonProps}>
```

Comment:
@lizozom had an idea about this - if I remember correctly, her idea was that this shouldn't be a button at all, but a `EuiMenuItem` with a background.
cc-ing Liza here as she knows the `top_nav_menu` component best.

Reply:
That sounds like a promising way to maybe avoid the height issue!

Reply:
Yeah, ping me tomorrow morning, I think there should be a much easier way to do this